### PR TITLE
Add specification doc

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2323

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# 1. Overview
+This library is based on [regular expressions](https://en.wikipedia.org/wiki/Regular_expression), which are notations 
+for describing sets of character strings that specify a search pattern.
+
+# 2. Operations
+
+## 2.1. Matches
+This is used to check whether a string matches the provided regex.
+```ballerina
+boolean isMatched = regex:matches("Ballerina is great", "Ba[a-z ]+");
```
Comment (daneshk) [inline docs/spec/spec.md:32]: Shall we mention add API definition instead of a sample

Comment (daneshk) [reply docs/spec/spec.md:32]: like https://github.com/ballerina-platform/module-ballerina-time/blob/master/docs/spec/spec.md

Comment (kaneeldias) [reply docs/spec/spec.md]: Fixed in c823587

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**